src/common/templates/_virtualservice.tpl

```diff
@@ -0,0 +1,67 @@
+{{/*
+USAGE:
+{{- include "harnesscommon.v1.renderVirtualService" (dict "ctx" $) }}
+*/}}
+{{- define "harnesscommon.v1.renderVirtualService" }}
+{{- $ := .ctx }}
+{{- if $.Values.global.istio.enabled -}}
+{{- range $index, $object := $.Values.virtualService.objects }}
+{{- $objName := dig "name" ((cat (default $.Chart.Name $.Values.nameOverride | trunc 63 | trimSuffix "-") "-" $index)| nospace)  $object }}
+apiVersion: networking.istio.io/v1beta1
+kind: VirtualService
+metadata:
+  name: {{ $objName }}
+  namespace: {{ $.Release.Namespace }}
+  {{- if $.Values.global.commonLabels }}
+  labels:
+    {{- include "harnesscommon.tplvalues.render" ( dict "value" $.Values.global.commonLabels "context" $ ) | nindent 4 }}
+  {{- end }}
+  annotations:
+    {{- include "harnesscommon.tplvalues.render" (dict "value" $object.annotations "context" $) | nindent 4 }}
+    {{- if $.Values.ingress.annotations }}
+    {{- include "harnesscommon.tplvalues.render" (dict "value" $.Values.virtualService.annotations "context" $) | nindent 4 }}
+    {{- end }}
+    {{- if $.Values.global.commonAnnotations }}
+    {{- include "harnesscommon.tplvalues.render" ( dict "value" $.Values.global.commonAnnotations "context" $ ) | nindent 4 }}
+    {{- end }}
+spec:
+  gateways:
+    {{- if and ($.Values.global.istio.virtualService.gateways) (eq (len $.Values.global.istio.virtualService.gateways) 0) }}
+    - istio-system/public
+    {{- else }}
+    {{- range $.Values.global.istio.virtualService.gateways }}
+    - {{ . }}
+    {{- end }}
+    {{- end }}
+  hosts:
+    {{- range $.Values.global.istio.virtualService.hosts }}
+    - {{ . }}
+    {{- end }}
+  http:
+  {{- range $i, $path := $object.paths }}
+  {{- $serviceName := dig "backend" "service" "name" $.Chart.Name $path }}
```

Comment:
lets add a way to override the serviceName and servicePort if possible. it should fallback to this method but sometimes service name are hardcoded like chaos-manager-service